pkg/clusterregistryconfig/flags.go

```diff
@@ -0,0 +1,349 @@
+				ocm.ValidateAllowedRegistriesForImport,
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid value for allowed registries for import: %s", err)
+		}
+
+		result.additionalTrustedCa, err = interactive.GetString(interactive.Input{
+			Question: "Registry Additional Trusted CA",
+			Help:     cmd.Lookup("registry-additional-trusted-ca").Usage,
+			Default:  args.additionalTrustedCa,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("Expected a valid certificate: %s", err)
+		}
+	}
+	if err := ocm.ValidateAllowedRegistriesForImport(result.allowedRegistriesForImport); err != nil {
+		return nil, fmt.Errorf("Expected valid allowed registries for import values: %v", err)
+	}
+
+	return result, nil
+}
+
+func IsClusterRegistryConfigSetViaCLI(cmd *pflag.FlagSet) bool {
+	for _, parameter := range []string{allowedRegistriesFlag,
+		insecureRegistriesFlag, blockedRegistriesFlag, platformAllowlistFlag,
+		allowedRegistriesForImportFlag, additionalTrustedCaFlag} {
+
+		if cmd.Changed(parameter) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func BuildRegistryConfig(args *ClusterRegistryConfigArgs) (*cmv1.ClusterRegistryConfigBuilder, error) {
+	isClusterRegistryConfigured := args.allowedRegistries != nil ||
+		args.blockedRegistries != nil || args.insecureRegistries != nil ||
+		args.allowedRegistriesForImport != "" || args.platformAllowlist != "" ||
+		args.additionalTrustedCa != ""
+
+	if isClusterRegistryConfigured {
+		registryResources := cmv1.NewRegistrySources()
+
+		if args.allowedRegistries != nil {
+			registryResources.AllowedRegistries(args.allowedRegistries...)
+		}
+		if args.blockedRegistries != nil {
+			registryResources.BlockedRegistries(args.blockedRegistries...)
+		}
+		if args.insecureRegistries != nil {
+			registryResources.InsecureRegistries(args.insecureRegistries...)
+		}
+
+		clusterRegistryConfig := cmv1.NewClusterRegistryConfig().
+			RegistrySources(registryResources)
+
+		if args.additionalTrustedCa != "" {
+			// read the file and make sure that it's in the right format
+			ca, err := BuildAdditionalTrustedCAFromInputFile(args.additionalTrustedCa)
+			if err != nil {
+				return nil, fmt.Errorf("Expected a valid value: %v", err)
+			}
+			// check if the ca is in "registryName: certificate" format
+			err = ocm.ValidateRegistryAdditionalCa(ca)
+			if err != nil {
+				return nil, fmt.Errorf("Expected a valid value: %v", err)
+			}
+			clusterRegistryConfig.AdditionalTrustedCa(ca)
+		}
+		if args.platformAllowlist != "" {
+			clusterRegistryConfig.PlatformAllowlist(
+				cmv1.NewRegistryAllowlist().ID(args.platformAllowlist))
+		}
+		if args.allowedRegistriesForImport != "" {
+			obj, err := BuildAllowedRegistriesForImport(args.allowedRegistriesForImport)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to build allowed registries for import, received error: %s", err)
+			}
+
+			//construct the location list
+			locationList := []*cmv1.RegistryLocationBuilder{}
```

Comment:
No need for this instantiation: all slice functions (`append`, `len`) are nil safe.
```suggestion
			var locationList []*cmv1.RegistryLocationBuilder
```

Reply:
Updated the var